### ISSUE TITLE
Standalone server requires geckodriver even with "marionette: false"

### ISSUE THREAD

**Issue by PyDeq** [SeleniumHQ/selenium#3757]
## Meta -
OS:  server: Debian 8 (jessie), client: openSUSE Leap 42.1
Selenium Version:  3.3.1
Browser:  Mozilla Firefox
Browser Version:  45.8.0

## Expected Behavior -
A `WebDriver` instance to be initialized.

## Actual Behavior -
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/tests/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 92, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/srv/tests/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 179, in start_session
    response = self.execute(Command.NEW_SESSION, capabilities)
  File "/srv/tests/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 238, in execute
    self.error_handler.check_response(response)
  File "/srv/tests/venv/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 193, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: The path to the driver executable must be set by the webdriver.gecko.driver system property; for more information, see https://github.com/mozilla/geckodriver. The latest version can be downloaded from https://github.com/mozilla/geckodriver/releases
Stacktrace:
    at com.google.common.base.Preconditions.checkState (Preconditions.java:738)
    at org.openqa.selenium.remote.service.DriverService.findExecutable (DriverService.java:111)
    at org.openqa.selenium.firefox.GeckoDriverService.access$100 (GeckoDriverService.java:38)
    at org.openqa.selenium.firefox.GeckoDriverService$Builder.findDefaultExecutable (GeckoDriverService.java:112)
    at org.openqa.selenium.remote.service.DriverService$Builder.build (DriverService.java:302)
    at org.openqa.selenium.firefox.FirefoxDriver.toExecutor (FirefoxDriver.java:233)
    at org.openqa.selenium.firefox.FirefoxDriver.<init> (FirefoxDriver.java:125)
    at org.openqa.selenium.firefox.FirefoxDriver.<init> (FirefoxDriver.java:150)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (NativeConstructorAccessorImpl.java:-2)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at org.openqa.selenium.remote.server.FirefoxDriverProvider.callConstructor (FirefoxDriverProvider.java:81)
    at org.openqa.selenium.remote.server.FirefoxDriverProvider.newInstance (FirefoxDriverProvider.java:63)
    at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance (DefaultDriverFactory.java:60)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:222)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:209)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at org.openqa.selenium.remote.server.DefaultSession$1.run (DefaultSession.java:176)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:617)
    at java.lang.Thread.run (Thread.java:745)
```

## Steps to reproduce -

```python
from selenium import webdriver

caps = webdriver.DesiredCapabilities.FIREFOX.copy()
caps["marionette"] = False
webdriver.Remote("http://turris-qa-lan-tester:4444/wd/hub", caps)
```

## Additional info -

It seems that this bug was introduced in Selenium 3.3.0 because Selenium 3.3.0 fails as well while Selenium 3.2.0 works just fine.
I guess that the root cause is that the server does not set the "legacy" flag if it gets `marionette: false` because I did not find a code (in 3.3.1) that calls `setLegacy` in this case and because setting the `webdriver.firefox.marionette` system property to `false` works as a workaround.

**Comment by AutomatedTester** [comment]
I believe that this was fixed in 3.3.1. Please can you upgrade and retest.

**Comment by PyDeq** [comment]
Unfortunately, as I said above, 3.3.1 suffers from this as well.

**Comment by lmtierney** [comment]
This was fixed in 14ef7bd and will be in the next release

**Comment by PyDeq** [comment]
I can confirm that it works for me in Selenium 3.4. Thank you!
